public/app/features/alerting/unified/rule-list/filter/RulesFilter.v2.tsx

```diff
@@ -97,6 +115,39 @@ export default function RulesFilter({ viewMode, onViewModeChange }: RulesFilterP
     setValue('query', searchQuery);
   }, [searchQuery, setValue]);
 
+  // Apply saved search - updates search input, triggers filtering, and updates URL
+  const handleApplySearch = useCallback(
+    (search: SavedSearch) => {
+      // Update the search input
+      setValue('query', search.query);
+
+      // Trigger the filtering mechanism
+      const parsedFilter = getSearchFilterFromQuery(search.query);
+      updateFilters(parsedFilter);
+
+      // Update URL with the search query for shareability
+      const currentParams = locationService.getSearchObject();
+      locationService.partial({ ...currentParams, search: search.query || undefined });
+
+      // Track analytics
+      trackSavedSearchApplied(search);
+    },
+    [setValue, updateFilters]
+  );
+
+  // Auto-apply default search on navigation (not refresh)
+  // The getAutoApplySearch() hook handles its own state to prevent double-apply
+  useEffect(() => {
+    if (!savedSearchesEnabled || savedSearchesLoading) {
+      return;
+    }
+
+    const defaultSearch = getAutoApplySearch();
```

Comment:
This looks a bit fishy. Why the double apply can happen? I wonder if we could come up with a better condition for applying the default search params